homeassistant/components/tasmota/mixins.py

```diff
@@ -25,7 +26,55 @@ class TasmotaEntity(Entity):
 
     def __init__(self, tasmota_entity) -> None:
         """Initialize."""
+        self._state = None
         self._tasmota_entity = tasmota_entity
+        self._unique_id = tasmota_entity.unique_id
+
+    async def async_added_to_hass(self):
+        """Subscribe to MQTT events."""
+        self._tasmota_entity.set_on_state_callback(self.state_updated)
+        await self._subscribe_topics()
+
+    async def async_will_remove_from_hass(self):
+        """Unsubscribe when removed."""
+        await self._tasmota_entity.unsubscribe_topics()
```

Comment:
Maybe you could have `self._subscribe_topics` already add this method as a callback for `self.async_on_remove` ?

Reply:
`self._tasmota_entity.unsubscribe_topics` is a coroutine function though, it's not compatible with `Entity.async_on_remove` :
```py
@callback
    def async_on_remove(self, func: CALLBACK_TYPE) -> None:
```

Reply:
`functools.partial(hass.async_create_task, self._tasmota_entity.unsubscribe_topics)`

Reply:
Will leave as is as discussed on Discord.